docs/src/main/paradox/server/akka-http.md

```diff
@@ -50,18 +50,157 @@ Java
 Finally we can combine the routes and serve them. Remember we need to use `bindAndHandleAsync` to enable HTTP/2 support:
 
 Scala
-:  @@snip [AuthenticatedGreeterServer.java](/plugin-tester-scala/src/main/scala/example/myapp/helloworld/AuthenticatedGreeterServer.scala) { #combined }
+:  @@snip [AuthenticatedGreeterServer.scala](/plugin-tester-scala/src/main/scala/example/myapp/helloworld/AuthenticatedGreeterServer.scala) { #combined }
 
 Java
 :  @@snip [AuthenticatedGreeterServer.java](/plugin-tester-java/src/main/java/example/myapp/helloworld/AuthenticatedGreeterServer.java) { #combined }
 
-### Future work
 
-For more in-depth integration you might want to pass information from the
-Akka HTTP route into your gRPC service implementation.
+## Example: logging, error handling, and passing request context
 
-Currently, you could achieve this by adding the required information to your
-service implementation constructor, and constructing a new Handler for each request.
+This example shows how we can wrap a gRPC service implementation into a @apidoc[Route$] to get the following common server features:
+
+1. Log the HTTP request and response corresponding to each RPC.
+2. Pass the @apidoc[RequestContext] into the RPC handler. 
+3. If the RPC fails, apply a custom error handler and log the error.
+
+### Implementation
+
+We start with an implementation of the sayHello RPC that does some primitive validation on the name.  
+If the name starts with a lowercase later, we return an `IllegalArgumentException`.
+Otherwise, we return a standard response.
+
+However, there's also a slight bug in the implementation.
+If the name is empty, the call to get the first character in the first if statement will throw an Exception.
+
+Scala
+:  @@snip [LoggingErrorHandlingGreeterServer.scala](/plugin-tester-scala/src/main/scala/example/myapp/helloworld/LoggingErrorHandlingGreeterServer.scala) { #implementation }
+
+Java
+:  @@snip [LoggingErrorHandlingGreeterServer.java](/plugin-tester-java/src/main/java/example/myapp/helloworld/LoggingErrorHandlingGreeterServer.java) { #implementation }
+
+### Method to log, handle, and recover each RPC 
+
+We define a method that returns a @apidoc[Route$] implementing our logging and error recovery features.
+The method takes three parameters.
+
+1. A function that takes a @apidoc[RequestContext] and returns a service implementation. This gives us the opportunity to use the context in the implementation. If we don't need it, we can just ignore the context and return a fixed implementation.
+2. A function that takes an @apidoc[ActorSystem] and returns a partial function from Throwable to gRPC @apidoc[Trailers].
+3. A function that takes the service implementation and an error handler and returns a request handler (a function from @apidoc[HttpRequest] to a Future of @apidoc[HttpResponse]). 
```

Comment:
For Java it's not Future, but probably a CompletionStage?
```
@scala[`Future`]@java[`CompletionStage`]
```

Reply:
Thanks, Fixed in [3421183](https://github.com/akka/akka-grpc/pull/1618/commits/342118367c0c32d9119ee3a6a1c9173c21a9bb73)